client-js/app/elements/labrad-registry.ts

```diff
@@ -129,6 +129,101 @@ export class LabradRegistry extends polymer.Base {
     );
   }
 
+  //Drag and drop
+  @listen('dragenter')
+  overBehaviour(event) {
+    event.preventDefault(); //I don't understand this.
+  }
+
+  @listen('dragover')
+  changeBehaviour(event) {
+    //allows the ctrl key to be pressed to change cursor between copy/move
+    event.preventDefault(); //I don't understand this. But needs to be here
+
+    if (event.ctrlKey) {
+      event.dataTransfer.dropEffect = "copy";
+    }
+    else {
+      event.dataTransfer.dropEffect = "move";
+    }
+  }
+
+  @listen('dragleave')
+  leaveDrag(event) {
+    event.preventDefault(); //I don't understand this. But it needs to be here
+  }
+
+  @listen('dragstart')
+  startDrag(event) {
+    //detect start of drag event, grab info about target
```

Comment:
We haven't been very consistent about format of comments, but the styleguide recommends JSDoc format (https://google.github.io/styleguide/javascriptguide.xml?showone=Comments#Comments) which puts the comments before the method and uses `/** */ as delimiters:
```typescript
/**
 * Detect start of drag event and grab info about target.
 */
@listen('dragstart')
startDrag(event) {
  ...
}
```